internal/controller/ocirepository_controller.go

```diff
@@ -672,26 +682,106 @@ func (r *OCIRepositoryReconciler) verifySignature(ctx context.Context, obj *ociv
 				SubjectRegExp: match.Subject,
 			})
 		}
-		defaultCosignOciOpts = append(defaultCosignOciOpts, soci.WithIdentities(identities))
+		defaultCosignOciOpts = append(defaultCosignOciOpts, scosign.WithIdentities(identities))
 
-		verifier, err := soci.NewCosignVerifier(ctxTimeout, defaultCosignOciOpts...)
+		verifier, err := scosign.NewCosignVerifier(ctxTimeout, defaultCosignOciOpts...)
 		if err != nil {
-			return err
+			return soci.VerificationResultFailed, err
 		}
 
 		signatures, _, err := verifier.VerifyImageSignatures(ctxTimeout, ref)
 		if err != nil {
-			return err
+			return soci.VerificationResultFailed, err
 		}
 
 		if len(signatures) > 0 {
-			return nil
+			return soci.VerificationResultSuccess, nil
 		}
 
-		return fmt.Errorf("no matching signatures were found for '%s'", ref)
+		return soci.VerificationResultFailed, fmt.Errorf("no matching signatures were found for '%s'", ref)
+
+	case "notation":
+		// get the public keys from the given secret
+		secretRef := obj.Spec.Verify.SecretRef
+
+		if secretRef == nil {
+			return soci.VerificationResultFailed, fmt.Errorf("verification secret cannot be empty: '%s'", ref)
+		}
+
+		verifySecret := types.NamespacedName{
+			Namespace: obj.Namespace,
+			Name:      secretRef.Name,
+		}
+
+		pubSecret, err := r.retrieveSecret(ctxTimeout, verifySecret)
+		if err != nil {
+			return soci.VerificationResultFailed, err
+		}
+
+		data, ok := pubSecret.Data[notation.DefaultTrustPolicyKey]
+		if !ok {
+			return soci.VerificationResultFailed, fmt.Errorf("'%s' not found in secret '%s'", notation.DefaultTrustPolicyKey, verifySecret.String())
+		}
+
+		var doc trustpolicy.Document
+
+		if err := json.Unmarshal(data, &doc); err != nil {
+			return soci.VerificationResultFailed, fmt.Errorf("error orrcured while parsing %s: %w", notation.DefaultTrustPolicyKey, err)
+		}
+
+		var certs [][]byte
+
+		for k, data := range pubSecret.Data {
+			if strings.HasSuffix(k, ".crt") || strings.HasSuffix(k, ".pem") {
+				certs = append(certs, data)
+			}
+		}
+
+		if certs == nil {
+			return soci.VerificationResultFailed, fmt.Errorf("no certificates found in secret '%s'", verifySecret.String())
+		}
+
+		trustPolicy := notation.CleanTrustPolicy(&doc, ctrl.LoggerFrom(ctx))
+		defaultNotationOciOpts := []notation.Options{
+			notation.WithTrustStore(trustPolicy),
+			notation.WithRemoteOptions(opt...),
+			notation.WithAuth(auth),
+			notation.WithKeychain(keychain),
+			notation.WithInsecureRegistry(obj.Spec.Insecure),
+			notation.WithLogger(ctrl.LoggerFrom(ctx)),
+		}
```

Comment:
```suggestion
		defaultNotaryOciOpts := []notation.Options{
			notation.WithTrustStore(trustPolicy),
			notation.WithRemoteOptions(verifyOpts...),
			notation.WithAuth(clientOpts.Authenticator),
			notation.WithKeychain(clientOpts.Keychain),
			notation.WithInsecureRegistry(clientOpts.Insecure),
			notation.WithLogger(ctrl.LoggerFrom(ctx)),
			notation.WithRootCertificate(certs),
		}
```

Reply:
Resolved in [cfcf21a](https://github.com/fluxcd/source-controller/pull/1075/commits/cfcf21a7519d407b437ed0435e4a7ffd98b1c7ce).